### ISSUE TITLE
mscHack modules for VCV Rack v2?

### ISSUE THREAD

**Issue by SaccoSVD** [mschack/VCV-Rack-Plugins#106]
I loooove your mixers (your rock) and I'm sad to see they're not present in V2

Cheers!

**Comment by falkTX** [comment]
It is fixed in #105

I already have it running as part of https://github.com/DISTRHO/Cardinal/

**Comment by SaccoSVD** [comment]
Superb!!!!

I have no idea how to build code. Is your plan to make them available again in the VCV store?

**Comment by falkTX** [comment]
I am not the author, that is not up to me.
But at least we know it is possible to make it work. It is up to the original authors to decide if they want to keep working on the plugins or not.

**Comment by SaccoSVD** [comment]
Got it. Thanks :)

**Comment by SaccoSVD** [comment]
> I am not the author, that is not up to me. But at least we know it is possible to make it work. It is up to the original authors to decide if they want to keep working on the plugins or not.

What did you use to compile the plugins to use in Rack v2? (is it even possible to use in the official rack v2 version?)

I have Visual Studio, Git, Git Desktop and I have only tried building OBS from a tutorial :) So now I see the files in this repo and I have no idea how to compile them so some clues would be very welcome, thank you. :)

**Comment by falkTX** [comment]
I dont use windows, but linux. On linux these tools are pretty much a single command away from being installed.
```
sudo apt-get install build-essential git
```
On Windows I guess you need msys2, then the "Rack SDK", clone the repo and place the files in the appropriate dir for things to build. There should be some official VCV docs about this somewhere...

**Comment by SaccoSVD** [comment]
So I followed all the instructions, have the SDK, and Msys and all environment variables added.

If I do:

make -o Mixer_24_4_4.cpp Mixer_24_4_4
or:
g++ Mixer_24_4_4.dll Mixer_24_4_4.cpp

I get:

g++     Mixer_24_4_4.cpp   -o Mixer_24_4_4
In file included from Mixer_x_x_x.hpp:1,
                 from Mixer_24_4_4.cpp:19:
mscHack.hpp:1:10: fatal error: rack.hpp: No such file or directory
    1 | #include "rack.hpp"
      |          ^~~~~~~~~~
compilation terminated.
make: *** [<builtin>: Mixer_24_4_4] Error 1



I'm sure I'm not doing something right. I know rack hpp is part of the SDK but make (or g++) is not able to locate it.

I'm not sure what's the correct syntax and also I'm not sure the extension of the plugin once compiled.

Yeah, I'm sorry, I'm super newb to this.

**Comment by falkTX** [comment]
err that is not how you build things. just "make" on the right dir with the right files in place typically does it, but then I dont know how the 2nd step goes (for making an actual vcv archive) as I never needed it.

I find the way to learn is just to do it yourself.
Study how a plugin is made, and try to do one.
otherwise you are just going blind at this. even if it works in the end, you wont know really how it worked, so assuming there are issues with mscHack (and there are a few yes) you wont be able to fix them either.

**Comment by SaccoSVD** [comment]
Yeah I knew I wasn't doing things right. Seems quite complicated at this point.

Doing just "make" in a CMD in the mschack folder where the Makefile is returns this:

G:\PRODUCCION\GitHub\mscHack\VCV-Rack-Plugins>make
Makefile:7: ../../plugin.mk: No such file or directory
make: *** No rule to make target '../../plugin.mk'.  Stop.

I do have a RACK_DIR environment variable installed with the right path (C:\Programas\msys2\Rack-SDK)

EDIT: I could start the compiling process by defining the SDK path manually in the Makefile. But it throws errors. So I give up as I have no clue what the messages are.

I only hope the author of these fine plugins wants to update them...in the meantime I guess I need a replacement...the shitty part is all my patches use the mixer from this collection and reconnecting everything is gonna be a PITA. (and no Rack v1 is not an option as is quite buggy when it comes to audio devices)

**Comment by SteveRussell33** [comment]
Follow the instructions here: https://vcvrack.com/manual/Building#Building-Rack-plugins

**Comment by SaccoSVD** [comment]
I did....no idea how it really works or what to expect, or how to fix potential errors. Is that something I need to do cause the built plugins are gonna be private?......or....can someone please make a windows build for us?
